### PR TITLE
Fix incorrect usage of CString because it didn't come from us

### PR DESCRIPTION
**Pull request by hbina**
According to the docs, `CString` should only be used if it originally came from our own Rust code. If the C-string comes from the C side, we need to use `CStr`.

Before,

```shell
hbina@akarin ~/g/uutils (hbina-fix-statfs-invalid-free)> git checkout main 
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
hbina@akarin ~/g/uutils (main)> cargo run --quiet -- df
free(): invalid pointer
fish: Job 1, 'cargo run --quiet -- df' terminated by signal SIGABRT (Abort)
```

After,

```shell
hbina@akarin ~/g/uutils (main) [1]> git checkout hbina-fix-statfs-invalid-free 
Switched to branch 'hbina-fix-statfs-invalid-free'
hbina@akarin ~/g/uutils (hbina-fix-statfs-invalid-free)> cargo run --quiet -- df
Filesystem          1K-blocks         Used    Available  Use% Mounted on       
udev                 16103256            0     16103256    0% /dev            
tmpfs                 3227756         2408      3225348    1% /run            
/dev/sda5           475656176    287543144    163881352   61% /               
tmpfs                16138760       448464     15690296    3% /dev/shm        
tmpfs                    5120            4         5116    1% /run/lock       
tmpfs                16138760            0     16138760    0% /sys/fs/cgroup  
/dev/loop0              56960        56960            0  100% /snap/core18/2284
/dev/loop2              63488        63488            0  100% /snap/core20/1361
/dev/loop3              63488        63488            0  100% /snap/core20/1376
/dev/loop1                128          128            0  100% /snap/bare/5    
/dev/loop5              66432        66432            0  100% /snap/gtk-common-themes/1514
/dev/loop6              44672        44672            0  100% /snap/snapd/14978
/dev/loop4             224256       224256            0  100% /snap/gnome-3-34-1804/77
/dev/loop7              56960        56960            0  100% /snap/core18/2344
/dev/loop8              55552        55552            0  100% /snap/snap-store/558
/dev/loop13            224256       224256            0  100% /snap/gnome-3-34-1804/66
/dev/loop10             44800        44800            0  100% /snap/snapd/15177
/dev/loop12             66816        66816            0  100% /snap/gtk-common-themes/1519
/dev/loop11            254848       254848            0  100% /snap/gnome-3-38-2004/99
/dev/loop14            253952       253952            0  100% /snap/gnome-3-38-2004/87
/dev/loop9              52352        52352            0  100% /snap/snap-store/518
/dev/sda1               98304        33528        64776   35% /boot/efi       
tmpfs                 3227752           80      3227672    1% /run/user/1000  
/dev/mmcblk0p1        3889728           16      3889712    1% /media/hbina/4D2D-4D72
```

Happened because I have some docker filesystems that requires privilege.

Signed-off-by: Hanif Ariffin <hanif.ariffin.4326@gmail.com>

### CODE REVIEW

File: src/uucore/src/lib/features/fsext.rs [hunk 1]
```diff
@@ -692,6 +692,8 @@ pub fn statfs<P: AsRef<Path>>(path: P) -> Result<StatFs, String>
 where
     Vec<u8>: From<P>,
 {
+    use std::ffi::CStr;
```
Comment (sylvestre) [inline src/uucore/src/lib/features/fsext.rs:695]:
```suggestion
```

File: src/uucore/src/lib/features/fsext.rs [hunk 2]
```diff
@@ -700,9 +702,11 @@ where
                     0 => Ok(buffer),
                     _ => {
                         let errno = IOError::last_os_error().raw_os_error().unwrap_or(0);
-                        Err(CString::from_raw(strerror(errno))
-                            .into_string()
-                            .unwrap_or_else(|_| "Unknown Error".to_owned()))
+                        let errmsg = CStr::from_ptr(strerror(errno))
```
Comment (sylvestre) [inline src/uucore/src/lib/features/fsext.rs:705]: can you ignore this word ? "errmsg" ?

Comment (hbina) [reply src/uucore/src/lib/features/fsext.rs]: I just inlined the return.

**Comment by sylvestre** [comment]
would it be possible to add a test or too complex?

**Comment by sylvestre** [comment]
@hbina ping?

**Comment by hbina** [comment]
I am not sure how to make `statfs` fail in a test.

**Pull request merged**